developer_manual/basics/controllers.rst

```diff
@@ -475,9 +475,14 @@ To migrate your API you can use the **OCP\\AppFramework\\OCSController** base cl
 
     }
 
-The format parameter works out of the box, no intervention is required.
+For ``OCSController`` classes and their methods, :ref:`responders <controller-responders>` can be registered as with any other ``Controller`` method.
+The ``OCSController`` class have however automatically two responders pre-installed:
+Both JSON (``application/json``) and XML (``text/xml``) are generated on-the-fly depending on the request by the browser/user.
+To select the output format, the ``?format=`` query parameter or the ``Accept`` header of the request work out of the box, no intervention is required.
+It is advised to prefer the header generally, as this is the more programmatic way.
 
-In order to make routing work for OCS routes you need to add a separate ‘ocs’ entry to the routing table of your app. Inside these are normal routes.
+In order to make routing work for OCS routes you need to add :ref:`a separate 'ocs' entry<routes_ocs>` to the routing table in ``appinfo/routes.php`` of your app.
```

Comment:
This should also mention the attributes you can use since a few versions and link to https://github.com/nextcloud/documentation/blob/de2b54eea51e510b55ff1e1d7d22f5f168eea5f3/developer_manual/basics/routing.rst#routing as an explanation of how to use both.

Reply:
Ok I saw you changed the routing explanation in a later commit.
I would still like to have attributes mentioned more clearly though.

Reply:
How about something like this?

```suggestion
To make routing work for OCS, the route must be registered in the core.
This can be made in two ways:
You can add an attribute `#[ApiRoute]` to the controller method.
Alternatively, you can add :ref:`a separate 'ocs' entry<routes_ocs>` to the routing table in ``appinfo/routes.php`` of your app.
```

Reply:
`This can be done in two ways` but other than that sounds good.